test/ctmrg/symmetrization.jl

```diff
@@ -0,0 +1,43 @@
+using Test
+using PEPSKit
+using PEPSKit: herm_depth, herm_width
+using TensorKit
+
+function _test_elementwise_equal(peps1::InfinitePEPS, peps2::InfinitePEPS)
+    return @test all(zip(peps1.A, peps2.A)) do (p1, p2)
+        p1.data == p2.data
+    end
+end
```

Comment:
Do we want to just define `==` for pepses? I am also not entirely convinced that unpacking the tensor data is a solid approach, that really is reaching into the tensor internals which might break at some point.

Reply:
Yes, that's definitely the better way to do it. I now defined `==` and `isapprox` methods for `InfinitePEPS` which just use the TensorKit methods element-wise.

Reply:
> On the other hand, I might be okay with getting that fixed later, I really want to invest some time in getting a good grip on all the different kwargs with some good automated options there, so that could also just fit in that PR.

Maybe I would suggest that we do it that way since it's not quite clear yet to me how to best automate all the options. Also, I'm always a bit hesitant to put functions into an algorithm struct since that makes it hard to save them to disk (which you probably want to do somewhere in the simulation workflow). So perhaps we can tackle that in a separate PR?

> I think we could probably have either heisenberg or j1j2 as a test, and invest some time in getting a proper examples thing set up to add both. From a CI point of view, I don't think that testing both really makes that big of a difference, except increase the waiting times

Sounds like a good idea. But would you in general like to keep the `square_lattice_heisenberg` and `square_lattice_j1j2` functions separate? Technically, the Heisenberg model is just a special case of the J1-J2 model. However, I thought it would be useful to have both functions because with the Heisenberg one you can tweak $(J_x, J_y, J_z)$ whereas with the J1-J2 one you only tweak $(J_1, J_2)$ - of course you could combine all of that into a single function.

One reason to maybe keep both tests would be to check that the PEPS optimization runs fine for different parameter ranges of the same Hamiltonian (which is not always guaranteed I guess). If we decide to test only one of them it's probably best to test the J1-J2 model since that also includes next-nearest neighbor terms.

Reply:
I guess the bigger question is if the different parameter regimes would fail because of an implementation detail, or just because the algorithms are not suited. The CI really is just there to check that the code does what it is advertising, and it might be that the implementations are "perfect", and still the optimization does not converge.
If it does not take too long, it's completely fine to keep them in, but we should not be using the test-suite to explore too many models I guess. I like that J1J2 tests the next-nearest neighbor terms, so that actually increases "test coverage", but tweaking parameters does not :)